app/src/parser_impl.c

```diff
@@ -0,0 +1,873 @@
+    }
+    case UINT32: {
+        uint32_t tmp = 0;
+        CHECK_ERROR(_readUInt32(c, &tmp))
+        *value = (uint64_t)tmp;
+        break;
+    }
+    case UINT64: {
+        CHECK_ERROR(_readUInt64(c, value))
+        break;
+    }
+    default:
+        return parser_msgpack_int_type_expected;
+        break;
+    }
+
+    return parser_ok;
+}
+
+parser_error_t _readBinFixed(parser_context_t *c, uint8_t *buff, uint16_t bufferLen)
+{
+    uint8_t binType = 0;
+    uint8_t binLen = 0;
+    CHECK_ERROR(_readUInt8(c, &binType))
+    switch (binType)
+    {
+        case BIN8: {
+            CHECK_ERROR(_readUInt8(c, &binLen))
+            break;
+        }
+        case BIN16:
+        case BIN32: {
+            return parser_msgpack_bin_type_not_supported;
+            break;
+        }
+        default: {
+            return parser_msgpack_bin_type_expected;
+            break;
+        }
+    }
+
+    if(binLen != bufferLen) {
+        return parser_msgpack_bin_unexpected_size;
+    }
+    CHECK_ERROR(_readBytes(c, buff, bufferLen))
+    return parser_ok;
+}
+
+static parser_error_t _readBin(parser_context_t *c, uint8_t *buff, uint16_t *bufferLen, uint16_t bufferMaxSize)
+{
+    uint8_t binType = 0;
+    uint16_t binLen = 0;
+    CHECK_ERROR(_readUInt8(c, &binType))
+    switch (binType)
+    {
+        case BIN8: {
+            uint8_t tmp = 0;
+            CHECK_ERROR(_readUInt8(c, &tmp))
+            binLen = (uint16_t)tmp;
+            break;
+        }
+        case BIN16: {
+            CHECK_ERROR(_readUInt16(c, &binLen))
+            break;
+        }
+        case BIN32: {
+            return parser_msgpack_bin_type_not_supported;
+            break;
+        }
+        default: {
+            return parser_msgpack_bin_type_expected;
+            break;
+        }
+    }
+
+    if(binLen > bufferMaxSize) {
+        return parser_msgpack_bin_unexpected_size;
+    }
+
+    *bufferLen = binLen;
+    CHECK_ERROR(_readBytes(c, buff, *bufferLen))
+    return parser_ok;
+}
+
+parser_error_t _readBool(parser_context_t *c, uint8_t *value)
+{
+    uint8_t tmp = 0;
+    CHECK_ERROR(_readUInt8(c, &tmp))
+    switch (tmp)
+    {
+        case BOOL_TRUE: {
+            *value = 1;
+            break;
+        }
+
+        case BOOL_FALSE: {
+            *value = 0;
+            break;
+        }
+        default: {
+            return parser_msgpack_bool_type_expected;
+            break;
+        }
+    }
+    return parser_ok;
+}
+
+static parser_error_t _readAssetParams(parser_context_t *c, txn_asset_config *asset_config)
+{
+    size_t paramsSize = 0;
+    CHECK_ERROR(_readMapSize(c, &paramsSize))
+
+    CHECK_ERROR(_findKey(c, KEY_APARAMS_MANAGER))
+    CHECK_ERROR(_readBinFixed(c, asset_config->params.manager, sizeof(asset_config->params.manager)))
+    addItem(1);
+
+    CHECK_ERROR(_findKey(c, KEY_APARAMS_RESERVE))
+    CHECK_ERROR(_readBinFixed(c, asset_config->params.reserve, sizeof(asset_config->params.reserve)))
+    addItem(2);
```

Comment:
should add item be checked with `CHECK_ERROR`?
is there any chance that we eventually add and there is no space?